index.bs

```diff
@@ -78,4 +260,327 @@ interface StorageBucket {
+
+        1. [=/Resolve=] |p| with |dictionary|.
+
+1. Return |p|.
+
+</div>
+
+<aside class="note">
+
+The [=StorageBucket/quota value=] will be ignored if it exceeds the total amount of space available to the [=/storage shelf=], i.e. site.
+Its intended use is to keep a specific bucket from using up the entire site's storage space.
+
+</aside>
+
+<h3 id="storage-bucket-durability">Durability</h3>
+
+A [=/storage bucket=] has a <dfn for=StorageBucket>durability value</dfn>, a {{StorageBucketDurability}}.
+The user agent MAY initialize this value to "{{StorageBucketDurability/strict}}" or
+"{{StorageBucketDurability/relaxed}}". The user agent SHOULD ignore the [=StorageBucket/durability value=]
+for operations that otherwise specify durability behavior.
+
+<div algorithm>
+
+The <dfn method for="StorageBucket">durability()</dfn> method steps are:
+
+1. Let |p| be [=a new promise=].
+
+1. Let |bucket| be [=this=]'s [=/storage bucket=].
+
+1. If |bucket| has been removed, [=reject=] |p| with an {{InvalidStateError}}.
+
+1. Otherwise, [=/resolve=] |p| with |bucket|'s [=StorageBucket/durability value=].
+
+1. Return |p|.
+
+</div>
+
+<aside class="note">
+
+The durability reflects whether the user agent will prioritize performance or durability when completing operations on data in the bucket.
+Some operations, such as {{IDBTransaction}}s, can specify a durability which will override the [=StorageBucket/durability value=]. The default
+[=StorageBucket/durability value=] is left to the user agent. Durability can be specified when a bucket is initially created, but not changed later.
+The {{StorageBucket/durability()}} method should report what the user agent will actually do. The value is one of the following:
+
+: "{{StorageBucketDurability/strict}}"
+:: The user agent will consider that operations on data in the bucket are successful only after verifying that outstanding changes have been written to a persistent storage medium.
+: "{{StorageBucketDurability/relaxed}}"
+:: The user agent will consider that operations on data in the bucket are successful as soon as all outstanding changes have been written to the operating system, without subsequent verification.
+
+</aside>
+
+<h3 id="storage-bucket-expiration">Expiration</h3>
+
+A [=/storage bucket=] has an <dfn for="StorageBucket">expiration value</dfn>, which is either null or a [=duration=], initially null.
+Specifies the upper limit of a bucket lifetime.
+
+User agents MUST remove any bucket that has expired when {{StorageBucketManager/keys}} is called.
+User agents MUST remove a bucket that has expired when it is opened via {{StorageBucketManager/open()}}.
+User agents MAY clear buckets whose [=/bucket mode=] is `"best-effort"` before the
+specified timestamp when faced with storage pressure.
+
+<div algorithm>
+
+The <dfn method for="StorageBucket">setExpires(|expires|)</dfn> method steps are:
```

Comment:
The webidl for this is not in this PR, but it uses DOMHighResTimestamp.  I think according to this guidance we should instead be using wall clock with Moments.  The method should probably take a Date or Temporal.Instant:

https://w3c.github.io/hr-time/#sec-tools

Reply:
It looks like `Date` is not an IDL type. Would this be a `double` in WebIDL? (https://webidl.spec.whatwg.org/#idl-types)

Reading DOMHighResTimeStamp, it mentions it could used to represent a Moment [1][2]. Do you think this could apply here?

[1] https://w3c.github.io/hr-time/#sec-domhighrestimestamp
[2] https://w3c.github.io/hr-time/#moments-and-durations

Reply:
After some standards guidance, received recommendation is to use DOMHighResTimeStamp and to convert to wall clock moment internally.